packages/nextjs/hooks/scaffold-eth/useWalletBalance.ts

```diff
@@ -0,0 +1,30 @@
+import { useCallback, useEffect, useState } from "react";
+import { useBalance } from "wagmi";
+import { useAppStore } from "~~/services/store/store";
+
+export function useWalletBalance(address?: string) {
```

Comment:
Why the address arg is optional?

Reply:
Because `wagmi`'s `useAccount` can return an undefined address and `useBalance` is ok with using an undefined address. So it works correct

Reply:
I saw that it was working properly, but from the DX perspective is a bit confusing using the hook `useAccountBalance` that has a non-mandatory `address` arg.

But yeah, we need it to be this way to play nicely with wagmi.

Thanks!

Reply:
it's much better than redundant `if-else`'s, problems with conditional hooks, and so on :)